dialogue-core/src/test/java/com/palantir/dialogue/core/CautiousIncreaseAggressiveDecreaseConcurrencyLimiterTest.java

```diff
@@ -185,16 +186,39 @@ public void onSuccess_releasesSuccessfully(Behavior behavior) {
     }
 
     @Test
-    public void onSuccess_dropsIfResponseIndicatesQosOrError_host() {
+    public void onSuccess_dropsIfResponseIndicatesQosOrError_host_308() {
         CautiousIncreaseAggressiveDecreaseConcurrencyLimiter limiter = limiter(Behavior.HOST_LEVEL);
-        for (int code : new int[] {308, 503}) {
-            Response response = mock(Response.class);
-            when(response.code()).thenReturn(code);
+        Response response = mock(Response.class);
+        when(response.getFirstHeader(eq("Location"))).thenReturn(Optional.of("https://localhost"));
+        when(response.code()).thenReturn(308);
 
-            double max = limiter.getLimit();
-            limiter.acquire(LimitEnforcement.DEFAULT_ENABLED).get().onSuccess(response);
-            assertThat(limiter.getLimit()).as("For status %d", code).isCloseTo(max * 0.9, Percentage.withPercentage(5));
-        }
+        double max = limiter.getLimit();
+        limiter.acquire(LimitEnforcement.DEFAULT_ENABLED).get().onSuccess(response);
+        assertThat(limiter.getLimit()).as("For status %d", 308).isCloseTo(max * 0.9, Percentage.withPercentage(5));
+    }
+
+    @Test
+    public void onSuccess_successIfResponseIndicatesNonQos308() {
+        // This represents google chunked-upload APIs which respond '308 Resume Incomplete' to indicate
```

Comment:
https://cloud.google.com/storage/docs/performing-resumable-uploads

I must say I had trouble following the problem here and it looks like there's quite [complicated history](https://insanecoding.blogspot.com/2014/02/http-308-incompetence-expected.html?m=1) for 308s.